packages/melos/lib/src/common/utils.dart

```diff
@@ -205,6 +209,77 @@ class PromptException extends MelosException {
   String toString() => 'Was unable to prompt for input:\n$message';
 }
 
+class PersistentShell {
```

Comment:
This should live in its own file instead of in the utils file :)
I think we can place it in `common` too.